tests/test_utils/test_flash_attention.py

```diff
@@ -1,130 +1,257 @@
-import random
+import math
 
 import pytest
 import torch
 from einops import rearrange
 
 from colossalai.kernel.cuda_native.mha.flash_attn_2 import HAS_FLASH_ATTN
 from colossalai.kernel.cuda_native.mha.mem_eff_attn import HAS_MEM_EFF_ATTN
+from colossalai.kernel.cuda_native.mha.utils import Repad, SeqLenInfo, Unpad
 from colossalai.testing import clear_cache_before_run, parameterize
 
 if HAS_MEM_EFF_ATTN or HAS_FLASH_ATTN:
     from colossalai.kernel.cuda_native import ColoAttention
     from colossalai.kernel.cuda_native.scaled_softmax import AttnMaskType
 
 DTYPE = [torch.float16, torch.bfloat16, torch.float32]
-
-
-def baseline_attention(Z, N_CTX, H, q, k, v, sm_scale):
-    M = torch.tril(torch.ones((N_CTX, N_CTX), device="cuda"))
-    p = torch.matmul(q, k.transpose(2, 3)) * sm_scale
-    for z in range(Z):
-        for h in range(H):
-            p[:, :, M == 0] = float("-inf")
-    p = torch.softmax(p.float(), dim=-1).half()
-    ref_out = torch.matmul(p, v)
-    return ref_out
+FLASH_DTYPE = [torch.float16, torch.bfloat16]
+
+
+def attention_ref(q, k, v, attn_mask=None, causal=False, upcast=True, reorder_ops=False):
```

Comment:
Could you make this baseline function have the same parameters as `ColoAttention`'s `forward`